tests/unit_tests/udf_communication/peer_communication/test_payload_receiver.py

```diff
@@ -168,6 +170,21 @@ def test_is_ready_to_stop_after_received_payload_in_reverse_sequence(number_of_m
     assert is_ready_to_stop
 
 
+@pytest.mark.parametrize("number_of_messages, duplicated_message",
+                         [(i, j) for i in range(1, 10) for j in range(0, i)])
+def test_is_ready_to_stop_payload_in_sequence_multiple_times(number_of_messages: int, duplicated_message: int):
+    test_setup = create_test_setup()
+    for sequence_number in range(number_of_messages):
+        message, frames = create_payload_message(test_setup, sequence_number)
+        test_setup.payload_receiver.received_payload(message, frames)
+    test_setup.reset_mock()
+    sequence_number = duplicated_message
+    message, frames = create_payload_message(test_setup, sequence_number)
```

Comment:
You only create a single "duplicated" message per test. 
Wouldn't it be better to create all duplicated messages here?

Reply:
Maybe, for the case I wanted to test, a single duplicated is enough.

Reply:
Changed it to duplicate all until duplicate idx